### PR TITLE
fix: support for single quotes in attributes of openml

### PR DESCRIPTION
**Pull request by lalo**
https://www.openml.org/api/v1/json/data/features/720

Let me know if this is the correct fix for this type of dataset.

### CODE REVIEW

File: coba/pipes/readers.py [hunk 1]
```diff
@@ -165,8 +165,13 @@ def filter(self, source: Iterable[str]) -> Iterable[MutableMap]:
 
             if attribute_match:
                 attribute_text  = attribute_match.group(1).strip()
```
Comment (lalo) [inline coba/pipes/readers.py:167]: the thing I'm not sure of here is that why are we splitting this string instead of pulling it directly from the json (https://www.openml.org/api/v1/json/data/features/720)?

Comment (mrucker) [reply coba/pipes/readers.py:167]: Yeah, so the original reason was because I had some problems early on with the Rest API features in Json not always matching up with the arff data. That was a long long time ago, and I don't actually know if it is true but it has just left me with a preference for ARFF, and I've slowly been trying to move the openml code over to just arff so we have to debug less. 

Also, ARFF is its own format outside of openml so it seemed to me like we should have correct support for it in general. I'm not sure how crazy I am about replacing the spaces with underscores just because I'm not sure it is necessary. Yeah that 720 dataset has a space in the attribute. In my opinion the real problem with ARFF is that it doesn't define whether users have to use double or single quotes so it is kind of tricky to parse. For example, it looks like line 169 in your code gets rid of the single quotes which is right in this case but it could have just as well been a double quote. I've even seen datasets in openml that have double quotes in the header and single quotes in the data (40985).

My approach so far has simply been to add a new unit test every time I find a weird ARFF edge case and I thought we'd eventually have them all. Also, there isn't any great light-weight ARFF parser. The most reputable one I've found is scipy but that seems awfully heavy handed to require an entire scipy dependency just to read ARFF data. The Openml python project uses liac-arff but that seems out of date and makes even heavier use of regex than we do.

All that to say, I've spent an unreasonable amount of time trying to figure out this ARFF problem :). My vote would be we add a unit test for attribute names with spaces (using both single quote and double quote) and then simply handle it as simply as possible. Thoughts?

Comment (mrucker) [reply coba/pipes/readers.py:167]: (Also just to be clear I didn't mean this particular ARFF problem. I just meant what to do about the ARFF format in general.)
